nima/testing/junit5/webserver/src/main/java/io/helidon/nima/testing/junit5/webserver/HelidonServerJunitExtension.java

```diff
@@ -201,25 +203,47 @@ private void setupServer(WebServer.Builder builder) {
         });
     }
 
-    private void addRouting(WebServer.Builder builder) {
+    private void addRouting(ServerConfig.Builder builder) {
+        Map<String, ListenerConfig.Builder> listenerConfigs = new HashMap<>();
+        Map<String, Router.Builder> routerBuilders = new HashMap<>();
+
+        listenerConfigs.put(DEFAULT_SOCKET_NAME, ListenerConfig.builder().from(builder));
+
         withStaticMethods(testClass, SetUpRoute.class, (setUpRoute, method) -> {
             // validate parameters
             String socketName = setUpRoute.value();
-            boolean isDefaultSocket = socketName.equals(DEFAULT_SOCKET_NAME);
 
             SetUpRouteHandler methodConsumer = createRoutingMethodCall(method);
 
-            extensions.forEach(it -> builder.socket(socketName,
-                                                    (socket, route) -> it.updateListenerBuilder(socketName,
-                                                                                                socket,
-                                                                                                route)));
+            ListenerConfig.Builder socketBuilder = listenerConfigs.computeIfAbsent(socketName, it -> ListenerConfig.builder());
+            Router.RouterBuilder<?> route = routerBuilders.computeIfAbsent(socketName, it -> Router.builder());
+
+            extensions.forEach(it -> it.updateListenerBuilder(socketName,
+                                                              socketBuilder,
+                                                              route));
+
+            methodConsumer.handle(socketName, builder, socketBuilder, route);
+        });
+
+        routerBuilders.forEach((socketName, routerBuilder) -> {
+            if (DEFAULT_SOCKET_NAME.equals(socketName)) {
+                builder.addRoutings(routerBuilder.build().routings());
+            } else {
+                listenerConfigs.computeIfAbsent(socketName, it -> ListenerConfig.builder())
+                        .addRoutings(routerBuilder.build().routings());
+            }
+        });
 
-            if (isDefaultSocket) {
-                builder.defaultSocket(socketBuilder -> {
-                    methodConsumer.handle(socketName, builder, socketBuilder, builder);
-                });
+        listenerConfigs.forEach((socketName, listenerBuilder) -> {
+            if (DEFAULT_SOCKET_NAME.equals(socketName)) {
```

Comment:
equalsIgnoreCase seems better for this

Reply:
we dod not use ignore case comparison for names, so it does not belong here